lib/mongo/crypt/auto_encrypter.rb

```diff
@@ -71,9 +73,52 @@ def setup_encrypter(options = {})
         @mongocryptd_client = Client.new(
                                 @encryption_options[:mongocryptd_uri],
                                 monitoring_io: mongocryptd_client_monitoring_io,
+                                server_selection_timeout: mongocryptd_server_selection_timeout, # For testing purposes
                               )
 
-        # TODO: use all the other options for auto-encryption/auto-decryption
+        # Tests fail with live background threads if the @mongocryptd client is not closed at the
+        # end of every test run. Better to write one line registering it as a local client than to
+        # do it separately in every test.
+        ClientRegistry.instance.register_local_client(@mongocryptd_client) if defined?(ClientRegistry)
```

Comment:
Can you please make a ticket to investigate why this is needed and preferably remove it before release?

Reply:
https://jira.mongodb.org/browse/RUBY-2069